addons/mail/static/src/js/services/chat_manager.js

```diff
@@ -1115,19 +1114,97 @@ var ChatManager =  AbstractService.extend({
             self.mentionPartnerSuggestions = result.mention_partner_suggestions;
             self.discussMenuID = result.menu_id;
 
-            // Shortcodes: canned responses and emojis
+            // Shortcodes: canned responses            
             _.each(result.shortcodes, function (s) {
-                if (s.shortcode_type === 'text') {
-                    self.cannedResponses.push(_.pick(s, ['id', 'source', 'substitution']));
-                } else {
-                    self.emojis.push(
-                        _.pick(s, ['id', 'source', 'unicode_source', 'substitution', 'description'])
-                    );
-                    self.emojiSubstitutions[_.escape(s.source)] = s.substitution;
-                    if (s.unicode_source) {
-                        self.emojiSubstitutions[_.escape(s.unicode_source)] = s.substitution;
-                        self.emojiUnicodes[_.escape(s.source)] = s.unicode_source;
-                    }
+                self.cannedResponses.push(_.pick(s, ['id', 'source', 'substitution']));
+            });
+            // emojis
+
+            var lastadded = null;
+            self.emojiUnicodes={
```

Comment:
i don't think this needs to be a class variable.  I think you should put this on top of the file as a constant.